### ISSUE TITLE
Build error when custom collection is expecting glob to have `tags` key in front matter.

### ISSUE THREAD

**Issue by tannerdolby** [11ty/eleventy#1512]
**Describe the bug**
> TypeError: Cannot read property 'length' of undefined

When I forget to add the `tags` key to a blog post that is being added to a custom collection in `.eleventy.js`, I'm unable to build the site using my build command `npm run build`. 

It's not necessarily a bug, but something I wanted to bring up in case someone else encounters this same error. 

**To Reproduce**

1. Create a custom collection in `.eleventy.js` using `getFilteredByGlob`

```js
eleventyConfig.addCollection("posts", function(collection) {
        return collection.getFilteredByGlob("src/writing/*.md");
    });
```

2. Forget to add the `tags` key as front matter data to new blog posts ie `.md` files added to `/writing/`.
```yaml
---
title: Some blog post
date: 2020-10-10
---

Some blog content.
```
3. Run `npm run build`
4. Get Error: `TypeError: Cannot read property 'length' of undefined`

**Expected behavior**
Expect the site to successfully build after running `npm run build`

**Screenshots**

```
Problem writing Eleventy templates: (more in DEBUG output)
> Having trouble writing template: _site/writing/index.html

`TemplateWriterWriteError` was thrown
> (./src/_includes/layouts/writings.njk)
  Template render error: (/Users/TannerDolby/11ty-portfolio/src/_includes/layouts/nav.njk)
  TypeError: Cannot read property 'length' of undefined
```

**Environment:**

- OS and Version: macOS
- Eleventy Version: 0.11.0

**Additional context**
I wanted to open an issue for this so if anyone encounters a similar error, a quick google search will lead them here and help fix things.

**Comment by zachleat** [comment]
Hmm, it would be better to add a nicer error message here, no?

**Comment by tannerdolby** [comment]
@zachleat Yeah, I think adding a nicer error message would help make things more clear. Maybe having the error mention that it has to do with a custom collection or a template missing the `tags` key in front matter.

**Comment by tannerdolby** [comment]
Maybe something like this, where the `Template render error` content gives a bit more info on why things happened.

```
Problem writing Eleventy templates: (more in DEBUG output)
> Having trouble writing template: _site/writing/index.html

`TemplateWriterWriteError` was thrown
> (./src/_includes/layouts/writings.njk)
  Template render error: Custom collection cannot recognize template without appropriate 'tags' defined
  TypeError: Cannot read property 'length' of undefined
```

**Comment by tannerdolby** [comment]
@Ryuno-Ki What do you think about this?

**Comment by Ryuno-Ki** [comment]
I'm sorry, but I'm in catch-up mode currently. I leave this issue open as unread to come to it another day.

**Comment by tannerdolby** [comment]
> TypeError: Cannot read property 'length' of undefined

This is referring to the collection array having a `length` of undefined. When a template doesn't have a `tags` key in front matter, and you try to create a custom collection with that template using `collection.getFilteredGlob`. I looked into `TemplateWriter.js` and this seems to be where the error is generated:

https://github.com/tannerdolby/eleventy/blob/2f59c34a8a17571612258b2b37c84dd1f529e008/src/TemplateWriter.js#L287-L292

**Comment by zachleat** [comment]
Related to https://github.com/11ty/eleventy/issues/3467 also maybe https://github.com/11ty/eleventy/issues/246 (both already shipped)

**Comment by zachleat** [comment]
Is this one still valid? I think #3467 handled it!
